File: GameRanksClient/src/app/models/serverStructs/element/DeveloperStruct.ts
import {Developer} from "../../entity/Developer";
import {GameStruct} from "./GameStruct";

export class DeveloperStruct{
	developer: Developer;
	avgScore: number;
	rankTitle: string;
	games: GameStruct[];

	constructor(developer: Developer, avgScore: number, rankTitle: string, games: GameStruct[]){
		this.developer = developer;
		this.avgScore = avgScore;
		this.rankTitle = rankTitle;
		this.games = games;
	}
}

File: GameRanks/src/main/java/GameRanks/GameRanks/clientStruct/element/DeveloperStruct.java
package GameRanks.GameRanks.clientStruct.element;

import GameRanks.GameRanks.model.Developer;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class DeveloperStruct {
    private Developer developer;
    public double avgScore;
    public String rankTitle;    //gold,silver,bronze,normal
    public Iterable<GameStruct> games;
}
